deltacat/compute/compactor_v2/utils/exception_handler.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional
+from pyarrow.lib import ArrowException, ArrowInvalid
+from ray.exceptions import WorkerCrashedError, NodeDiedError, OutOfMemoryError, RayError
+from deltacat.exceptions import (
+    UploadTableThrottlingError,
+    DependencyRayWorkerDiedError,
+    DependencyRayError,
+    DependencyBotocoreError,
+    DependencyRayOutOfMemoryError,
+    DependencyPyarrowError,
+)
+from botocore.exceptions import ClientError as BotocoreClientError
+from deltacat.aws.constants import UPLOAD_TABLE_POSSIBLE_ERROR_CODE
+
+
+def handle_compaction_step_exception(e: Exception, task_id: Optional[str] = None):
```

Comment:
Check errors we are retrying here: https://github.com/ray-project/deltacat/blob/main/deltacat/compute/compactor_v2/utils/task_options.py#L72. Ultimately, all of them must be converted to `RetryableError` and we should edit that list to contain only `RetryableError`. This handle is responsible for letting Ray know whether to retry this task or not, and if yes, what is the new resource config.